### ISSUE TITLE
Feat: Depreciate git submodules usage and embrace `jj` or 'git branchless'

### ISSUE THREAD

**Issue by sambacha** [manifoldfinance/mevETH2#163]
# Feat: Depreciate git submodules usage and embrace `jj` or 'git branchless'

TLDR: git submodules take almost 3 mins to recursively clone. We want a solution to vendor such dependencies. 

I have spoken to @capybara23 about this issue on and off and the top contenders seem to be: 

## [jj](https://github.com/martinvonz/jj)

> https://github.com/martinvonz/jj

### slides for `jj`

[jj slides](https://docs.google.com/presentation/d/1F8j9_UOOSGUN9MvHxPZX_L4bQ9NMcYOp1isn17kTC_M/edit#slide=id.g152b1fb8869_0_1262)

#### Entire repo under source control

Refs, anonymous branches, working copy position in each worktree
- Think “transactional reflog” with extra metadata
Consequences: time travel! 🕟✈️
- View repo at a point in the past ✅
- Very useful for troubleshooting
- Restore repo to earlier state ✅
- Undo  🎉
`git revert` but on an operation
> Safe concurrency, even when distributed

## Git Branchless

https://github.com/arxanas/git-branchless

## Honorable Mention

[JOSH - https://github.com/josh-project/josh](https://github.com/josh-project/josh)

**Comment by sambacha** [comment]
@capybara23 says after oceansprint
